Objects/complexobject.c

```diff
@@ -256,26 +256,51 @@ PyComplex_FromDoubles(double real, double imag)
     return PyComplex_FromCComplex(c);
 }
 
+static PyObject * try_complex_special_method(PyObject *);
+
 double
 PyComplex_RealAsDouble(PyObject *op)
 {
+    double real = -1.0;
```

Comment:
I am thinking about whether "real" and "imag" below should return something else by default if op is not a valid object. Say something like NaN, given that -1.0 can be a valid real or imaginary part

Reply:
``1`` or ``1.0`` is a more common for C-API function choice, if they return int/long/double, see e.g. ``PyLong_AsLong()`` or ``PyLong_AsDouble()``.   This convention was also extended to ``PyComplex_AsCComplex()``.   And it was used before in ``PyComplex_RealAsDouble()``, so your suggestion will break backward compatibility.